src/generator/client.ts

```diff
@@ -96,31 +106,66 @@ export async function generateClient(session: Session<CodeModel>): Promise<strin
   if (session.model.info.description) {
     text += `// ${client} - ${session.model.info.description}\n`;
   }
+  // initialize clientOnlyParams in order to use them in client parameter declarations and assign values for the defaults later on
+  const clientOnlyParams = getGlobalClientOnlyParams(paramHostInfo.clientParams, session.model);
+  // initialize vars to add params specific to the client on the function signature and to pass
+  // params between functions
+  let clientOnlyParamsFuncSig = '';
+  let passClientOnlyParams = '';
+  // if the parameterized host functionality is not going to be implemented then wipe client only parameter settings
+  if (!paramHostInfo.addParamHost && !paramHostInfo.urlOnClient) {
+    clientOnlyParamsFuncSig = '';
+    passClientOnlyParams = '';
+  }
   text += `type ${client} struct {\n`;
-  text += `\t${urlVar} *url.URL\n`;
+  if (paramHostInfo.addParamHost && !paramHostInfo.urlOnClient) {
+    // add fields that are global client only params
+    for (const param of values(clientOnlyParams)) {
+      if (param.protocol.http!.in === 'uri') {
+        text += `\t${param.language.go!.name} ${param.schema.language.go!.name}\n`;
```

Comment:
So the reason for having the `!` is because by this point in our generator we know that the params will have `language.go` on them, since we've already performed transforms and naming changes in this field for all params. I'm open to changing to check this in the future however if you recommend it. It's sort of implicitly known for us at this point.